src/configuration.ts

```diff
@@ -0,0 +1,78 @@
+import * as vs from 'vscode';
+import * as fs from 'fs';
+import * as cp from 'child_process';
+import * as path from 'path';
+import Rubocop from './rubocop';
+
+export interface RubocopConfig {
+    command: string;
+    onSave: boolean;
+    configFilePath: string;
+    useBundler: boolean;
+}
+
+export const onDidChangeConfiguration: (rubocop: Rubocop) => () => void = (rubocop) => {
+    return () => rubocop.config = getConfig();
+};
+
+/**
+ * Read the workspace configuration for 'ruby.rubocop' and return a RubocopConfig.
+ * @return {RubocopConfig} config object
+ */
+export const getConfig: () => RubocopConfig = () => {
+    const cmd = (process.platform === 'win32') ? 'rubocop.bat' : 'rubocop';
+    const conf = vs.workspace.getConfiguration('ruby.rubocop');
+    let useBundler;
+    let path = conf.get('executePath', '');
+    let command;
+
+    // if executePath is present, use it.
+    if (path.length !== 0) {
+        command = path + cmd;
+    } else if (detectBundledRubocop()) {
+        useBundler = true;
+        command = `bundle exec ${cmd}`;
+        // alternately, use: `bundleShowRubocopOutput/bin/rubocop` ???
```

Comment:
2 thoughts here:
- It occurred to me that I could just detect the bundled path in place here- I'm not sure if there is a reason to run the command with `bundle exec` (my thought is yes, if that is the way the project/Gemfile is designed to be run), but the executable is right there.
- In writing tests, I ended up needing to stub bundler detection for both the `useBundler` and `command` property tests- maybe `command` doesn't need to worry about bundling and that can just be handled in the rubocop module itself (or not at all, as described above).

Reply:
@erikdstock One reason I can think of invoking `bundler exec` is in cases where people are using [rubocop extensions or formatters](https://github.com/bbatsov/rubocop/blob/master/manual/extensions.md) bundled in other gems or libraries. If you try to invoke the appropriate rubocop binary directly without the context of bundler, any config referencing such extensions/formatters would probably break.